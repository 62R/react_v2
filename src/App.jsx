// import { Form } from "./components/FormClass/Form"

import { Form } from './components/Form/Form';

export const App = () => {
  return (
    <div>
      <h1>React App</h1>
      <Form />
    </div>
  );
};
