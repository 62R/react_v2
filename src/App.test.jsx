import { act, render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { App } from './App';

describe('Input', () => {
  it('render component with spanshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('write message', async () => {
    render(<App />);
    const inputs = screen.getAllByRole('textbox');
    const button = screen.getByRole('button');
    await act(async () => {
      await userEvent.type(inputs[0], 'Jon');
      await userEvent.type(inputs[1], 'Text');
    });
    await act(async () => {
      await userEvent.click(button);
    });
    expect(screen.getByText('Text')).toBeInTheDocument();
  });

  it('Bot write answer after delay', async () => {
    render(<App />);
    const inputs = screen.getAllByRole('textbox');
    const button = screen.getByRole('button');
    await act(async () => {
      await userEvent.type(inputs[0], 'Jon');
      await userEvent.type(inputs[1], 'Text');
    });
    await act(async () => {
      await userEvent.click(button);
    });
    await waitFor(() => expect(screen.getByText('Ok')).toBeInTheDocument(), {
      timeout: 2000,
    });
  });
});
