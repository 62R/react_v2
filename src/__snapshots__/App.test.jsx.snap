// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render component with spanshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      React App
    </h1>
    <div
      class="wrapper"
    >
      <div
        class="messagelist"
      >
        <div
          class="message message_align_bot message_margin_otherauthor message_order_first"
        >
          <p
            class="author"
          >
             Bot 
          </p>
          <p
            class="text"
          >
             Wait a massage 
          </p>
        </div>
      </div>
      <form
        class="form"
      >
        <input
          class="input"
          placeholder="Enter your name"
          type="text"
          value=""
        />
        <textarea
          class="textarea"
          placeholder="Write some message"
        />
        <button
          class="button button_disabled"
          disabled=""
        >
          Push
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
